### PR TITLE
Gamemode Picking refactor

### PR DESCRIPTION
**Pull request by qualitykk**
Change TFGameRules to make the creation of custom gamemodes easier.
Now custom gamemodes are defined as classes and handle win checks.

### CODE REVIEW

**Review by Retr0-01 — APPROVED** [review]
LGTM 👍

File: code/Entities/Item.cs [hunk 1]
```diff
@@ -47,14 +48,19 @@ public virtual void Pickup( TFPlayer player )
 		player.PickedItem = this;
 	}
 
-	public virtual void Drop( TFPlayer player, bool dropped, bool message )
+	public virtual bool Drop()
 	{
-		player.PickedItem = null;
+		if ( !Game.IsServer ) return false;
+		if ( Owner == null ) return false;
```
Comment (rob5300) [inline code/Entities/Item.cs:54]: combine?

File: code/Entities/Objectives/Flag.cs [hunk 2]
```diff
@@ -205,16 +215,14 @@ public void Return()
...
+		EventDispatcher.InvokeEvent(new FlagReturnedEvent() { Flag = this } );
 	}
-
-	public override void Drop( TFPlayer player, bool dropped, bool message )
+	public override bool Drop()
 	{
-		if ( !Game.IsServer ) return;
-		if ( Disabled ) return;
+		if ( Disabled ) return false;
+		if ( !Drop() ) return false;
```
Comment (rob5300) [inline code/Entities/Objectives/Flag.cs:223]: Also combine

File: code/GameRules/Gamemodes/Arena.cs [hunk 3]
```diff
@@ -12,12 +13,42 @@ namespace TFS2;
...
+public partial class Arena : GamemodeEntity
 {
+	public override string Title => "Arena";
+
 	[Property, FGDType( "target_destination" )]
 	public string PointName { get; set; }
 	[Property] public float ControlPointEnableTime { get; set; } = 60;
-	ControlPoint Point { get; set; }
+	protected ControlPoint Point { get; set; }
+	public override GamemodeProperties Properties => new() { DisablePlayerRespawn = true, RequireBothTeams = true, ShouldAnnounceFirstBlood = true, ShouldPlayGameStartSong = false, AllowTeamSelection = false };
```
Comment (rob5300) [inline code/GameRules/Gamemodes/Arena.cs:24]: May be better to return a private instance instead of returning a new instance in the property itself.

**Review by MoonlyDays — APPROVED** [review]

**Review by rob5300 — APPROVED** [review]

**Pull request merged**